glide-core/src/client/value_conversion.rs

```diff
@@ -67,16 +96,92 @@ pub(crate) fn convert_to_expected_type(
             Value::Nil => Ok(value),
             _ => Ok(Value::Double(from_owned_redis_value::<f64>(value)?.into())),
         },
+        ExpectedReturnType::ArrayOfDouble => match value {
+            Value::Nil => Ok(value),
+            Value::Array(array) => {
+                let result: Result<Vec<Value>, _> = array
+                    .into_iter()
+                    .map(|element| match element {
+                        Value::BulkString(_) => {
+                            convert_to_expected_type(element, Some(ExpectedReturnType::Double))
+                        }
+                        _ => Ok(element),
+                    })
+                    .collect();
+
+                result.map(Value::Array)
+            }
+            _ => Err((
+                ErrorKind::TypeError,
+                "Response couldn't be converted to Array",
+                format!("(response was {:?})", value),
+            )
+                .into()),
+        },
+        ExpectedReturnType::BulkString => Ok(Value::BulkString(
+            from_owned_redis_value::<String>(value)?.into(),
+        )),
     }
 }
 
+fn convert_array_to_map(
+    array: Vec<Value>,
+    key_expected_return_type: Option<ExpectedReturnType>,
+    value_expected_return_type: Option<ExpectedReturnType>,
+) -> RedisResult<Value> {
+    let mut map = Vec::new();
+    let mut iterator = array.into_iter();
+    while let Some(key) = iterator.next() {
+        match key {
+            Value::Array(inner_array) => {
+                if inner_array.len() != 2 {
+                    return Err((
+                        ErrorKind::TypeError,
+                        "Array inside map must contain exactly two elements",
+                    )
+                        .into());
+                }
+                let mut inner_iterator = inner_array.into_iter();
+                let inner_key = inner_iterator.next().unwrap();
```

Comment:
It's preferable not to `unwrap` in production code, only in tests. If you're keeping the unwrap because the check above ensures it is safe, add a comment explaining this.